src/TraceEvent/DynamicTraceEventParser.cs

```diff
@@ -525,7 +526,7 @@ private object GetPayloadValueAt(ref PayloadFetch payloadFetch, int offset, int
                 }
 
                 // TODO this is very inefficient for blitable types. Optimize that.
-                var ret = Array.CreateInstance(elementType, arrayCount);
+                var ret = new object[arrayCount];
```

Comment:
This changes the dynamic return type of the API and likely breaks consumers. Although the return is statically typed as object I don't think it is really free to return an arbitrary type.

I believe arrayInfo.Element.Type is a small closed set of potential types so a helper method with a switch case could create this correctly.